pymatgen/command_line/bader_caller.py

```diff
@@ -80,44 +78,66 @@ def __init__(
             potcar_filename (str): The filename of the POTCAR.
             chgref_filename (str): The filename of the
                 reference charge density.
-            parse_atomic_densities (bool, optional): Enable atomic partition
-                of the charge density. Charge densities are atom centered.
-                Defaults to False.
             cube_filename (str, optional): The filename of the cube file.
-            bader_exe_path (str, optional): The path to the bader executable.
+            bader_path (str, optional): The path to the bader executable.
+            parse_atomic_densities (bool, optional): Enable atomic partition of the
+                charge density. Charge densities are atom centered. Defaults to False.
         """
-        bader_exe = which(bader_exe_path or "")
-        if bader_exe is None:
+
+        def temp_decompress(file: str | Path, target_dir: str = ".") -> str:
```

Comment:
is this guaranteed to make `temp` files? where are the decompressed files deleted?

Reply:
Hi thanks for asking and reviewing. This `temp_decompress` works more as an internal utility to work with `Scratch` (from monty.tempfile.Scratch) as described in its docstring

https://github.com/DanielYang59/pymatgen/blob/b3a989a37fd7e4e727410536a1a1f96077f1d940/pymatgen/command_line/bader_caller.py#L131-L161

As such the decompressed files should be created inside the `Scratch` dir and would be deleted once the context manager exists.

But I have a feeling this could be improved so let me know if you have any suggestion. Thanks! 😃

Reply:
me too. this code feels quite messy so i'm glad you're having a go at cleaning it up 👍

Reply:
> this code feels quite messy so i'm glad you're having a go at cleaning it up 👍

I actually wanted to say my `temp_decompress` could be improved 😄... But well... `BaderAnalysis` does have quite some code crammed inside `__init__` so cleaning it up a bit makes it easier to read. Would see if I could further make some improvement soon. Glad I could help.